### PR TITLE
Skip adding dimensions with empty dimension value into default dimension

### PR DESCRIPTION
**Pull request by paggynie**
*Issue #, if available:*
https://github.com/awslabs/aws-embedded-metrics-java/issues/125

*Description of changes:*
Customer with ECS FireLens now has problem to flush metrics logger when using default dimension because we return "" for log group name but we throw InvalidDimensionException when setting the empty dimension. Example stack trace

```
software.amazon.cloudwatchlogs.emf.exception.InvalidDimensionException: Dimension value cannot be empty
at software.amazon.cloudwatchlogs.emf.util.Validator.validateDimensionSet(Validator.java:48)
at software.amazon.cloudwatchlogs.emf.model.DimensionSet.addDimension(DimensionSet.java:167)
at software.amazon.cloudwatchlogs.emf.logger.MetricsLogger.configureContextForEnvironment(MetricsLogger.java:268)
at software.amazon.cloudwatchlogs.emf.logger.MetricsLogger.flush(MetricsLogger.java:91)
at com.amazonaws.toledo.telemetry.common.util.MetricsPublisher.flush(MetricsPublisher.java:39)
at com.amazonaws.toledo.telemetry.streamprocessor.StreamProcessor.processRecords(StreamProcessor.java:147)
at com.amazonaws.services.kinesis.clientlibrary.lib.worker.ProcessTask.callProcessRecords(ProcessTask.java:221)
at com.amazonaws.services.kinesis.clientlibrary.lib.worker.ProcessTask.call(ProcessTask.java:176)
at com.amazonaws.services.kinesis.clientlibrary.lib.worker.MetricsCollectingTaskDecorator.call(MetricsCollectingTaskDecorator.java:49)
at com.amazonaws.services.kinesis.clientlibrary.lib.worker.MetricsCollectingTaskDecorator.call(MetricsCollectingTaskDecorator.java:24)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:829)
```

This change is to skip adding dimension if dimension value is null or empty

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java [hunk 1]
```diff
@@ -259,19 +260,25 @@ public MetricsLogger setTimestamp(Instant timestamp) throws InvalidTimestampExce
...
+        setDefaultDimension(defaultDimension, "LogGroup", environment.getLogGroupName());
+        setDefaultDimension(defaultDimension, "ServiceName",  environment.getName());
+        setDefaultDimension(defaultDimension, "ServiceType", environment.getType());
         context.setDefaultDimensions(defaultDimension);
         environment.configureContext(context);
     }
 
+    @SneakyThrows
+    private void setDefaultDimension(DimensionSet defaultDimension, String dimKey, String dimVal) {
+        if (!StringUtils.isNullOrEmpty(dimVal)) {
```
Comment (markkuhn) [inline src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java:277]: Non-empty strings with only whitespace will still throw an error.
What about having an extra check for `isBlank()` ?

**Comment by markkuhn** [comment]
@paggynie Build is failing due to `spotless`. Please run `./gradlew :spotlessApply` to fix it.

File: src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java (hunk 1 above)
Comment (paggynie) [reply src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java:277]: I will just simply try on existing dimension validator as even we fix blank, another question can come out what if this dimensionValue has non valid ascii : )

File: src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java [hunk 2]
```diff
@@ -259,19 +260,26 @@ public MetricsLogger setTimestamp(Instant timestamp) throws InvalidTimestampExce
...
         environment.configureContext(context);
     }
 
+    private void setDefaultDimension(DimensionSet defaultDimension, String dimKey, String dimVal) {
+        try {
+            defaultDimension.addDimension(dimKey, dimVal);
+        } catch (InvalidDimensionException | DimensionSetExceededException e) {
+            // just do nothing
+        }
+    }
```
Comment (markkuhn) [inline src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java:281]: To follow convention/readability I would suggest changing it to:

```suggestion
        try {
            defaultDimension.addDimension(dimKey, dimVal);
        } catch (InvalidDimensionException | DimensionSetExceededException ignored) {}
    }
```

Comment (paggynie) [reply src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java]: tried but not spotless style. not sure there's any difference here

**Review by Himtanaya — APPROVED** [review]

File: src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java (hunk 2 above)
Comment (markkuhn) [reply src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java]: change the exception from `e` to `ignored`

**Review by markkuhn — APPROVED** [review]

**Comment by markkuhn** [comment]
🚀

**Pull request merged**